codeguru_profiler_agent/sdk_reporter/sdk_reporter.py

```diff
@@ -90,11 +112,65 @@ def report(self, profile):
             )
             logger.info("Reported profile successfully")
             return True
+        except ClientError as error:
+            if error.response['Error']['Code'] == 'ResourceNotFoundException':
+                if self.should_autocreate_profiling_group():
+                    is_create_pg_called_during_submit_profile = True
+                    logger.info("ResourceNotFoundException for a Lambda PG in PostAgentProfile call. " +
+                                "Attempting to create PG")
+                    self.create_profiling_group()
+            return False
         except Exception as e:
             self._log_request_failed(operation="post_agent_profile", exception=e)
             return False
 
+    @with_timer("createProfilingGroup", measurement="wall-clock-time")
+    def create_profiling_group(self):
+        """
+        Create a PG for the Lambda function onboarded with 1-click integration
```

Comment:
Change this comment. This will also be done if the customer doesn't use the code change option?

Reply:
Yes done